.github/workflows/workflow.yaml

```diff
@@ -317,6 +317,60 @@ jobs:
             done
           done
 
+  benchmark:
+    name: "Benchmark"
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Download release benchmark file
+        uses: robinraju/release-downloader@v1.6
+        id: get-latest-benchmark
+        continue-on-error: true
+        with: 
+          latest: true
+          fileName: "release_benchmarks.txt"
+
+      - name: Run benchmarks
+        run: |
+          curl -L -O "https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${KUBEBUILDER_VERSION}/kubebuilder_${KUBEBUILDER_VERSION}_linux_amd64.tar.gz" &&\
+            tar -zxvf kubebuilder_${KUBEBUILDER_VERSION}_linux_amd64.tar.gz &&\
+            sudo mv kubebuilder_${KUBEBUILDER_VERSION}_linux_amd64 /usr/local/kubebuilder
+          make benchmark-test BENCHMARK_FILE_NAME=pr_benchmarks.txt
+        if: ${{ steps.get-latest-benchmark.outcome == 'success' }}
+        env:
+          KUBEBUILDER_VERSION: 2.3.1
+
+      - name: Compare benchmarks
+        id: get-comment-body
+        if: ${{ steps.get-latest-benchmark.outcome == 'success' }}
+        run: |
+          export GOPATH="$HOME/go"
+          PATH="$GOPATH/bin:$PATH"
+          go install golang.org/x/perf/cmd/benchstat@latest
+          benchstat release_benchmarks.txt pr_benchmarks.txt > benchstat.txt
```

Comment:
what happens if `release_benchmarks.txt` doesn't exist?

Reply:
If the `release_benchmark.txt` doesn't exist `steps.get-latest-benchmark.outcome` results in `failure` instead of `success` and hence we skip the rest of the steps.